src/class/cdc/cdc_device.c

```diff
@@ -87,6 +87,24 @@ typedef struct
 //--------------------------------------------------------------------+
 CFG_TUSB_MEM_SECTION static cdcd_interface_t _cdcd_itf[CFG_TUD_CDC] = { { 0 } };
 
+bool pending_read_from_host;
+static void _prep_next_transaction(void) {

```

Comment:
we should have `itf` passed to this prepare function. Although it is assumed to zero for now in some functions, those are top-level (require a bit of code to detect the interface number).